sdk/identity/Azure.Identity/src/DefaultAzureCredential.cs

```diff
@@ -90,53 +92,80 @@ public override async ValueTask<AccessToken> GetTokenAsync(TokenRequestContext r
             return await GetTokenAsync(true, requestContext, cancellationToken).ConfigureAwait(false);
         }
 
-        private async Task<AccessToken> GetTokenAsync(bool isAsync, TokenRequestContext requestContext, CancellationToken cancellationToken)
+        private async ValueTask<AccessToken> GetTokenAsync(bool isAsync, TokenRequestContext requestContext, CancellationToken cancellationToken)
         {
             using CredentialDiagnosticScope scope = _pipeline.StartGetTokenScope("DefaultAzureCredential.GetToken", requestContext);
 
-            List<Exception> exceptions = new List<Exception>();
+            try
+            {
+                AccessToken token;
+
+                if (_credential != null)
+                {
+                    token = isAsync ? await _credential.GetTokenAsync(requestContext, cancellationToken).ConfigureAwait(false) : _credential.GetToken(requestContext, cancellationToken);
```

Comment:
Wonder if it's worth merging all the logic into `GetTokenFromSourcesAsync`.

Reply:
It doesn't really matter, but conceptually what I was doing is `GetTokenFromSources` handles walking the chain and finding a credential and this method is responsible for getting the Token and handling diagnostic scope. But, I don't really feel strongly either way, so if you really want me to change it I can do that 😉.